### PR TITLE
Add http proxy support for HttpClient

### PR DESCRIPTION
**Pull request by Praveen2112**

### CODE REVIEW

**Review by aczajkowski — COMMENTED** [review]
LGTM % small comments

File: http-client/src/test/java/io/airlift/http/client/jetty/AbstractHttpClientTestWithProxy.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        return new HttpClientConfig()
+                .setHttpProxy(testingHttpProxy.getHostAndPort());
+    }
+
+    @Override
+    @Test(enabled = false)
+    public void testConnectTimeout()
+            throws Exception
+    {
+        // todo jetty client does not timeout the socks proxy connect properly
```
Comment (electrum) [inline http-client/src/test/java/io/airlift/http/client/jetty/AbstractHttpClientTestWithProxy.java:79]: The "socks" here seems wrong

**Comment by Praveen2112** [comment]
@electrum AC

**Comment by electrum** [comment]
Thanks!

**Pull request merged**